lib/ma_crossover/events/data_managed_candles.js

```diff
@@ -89,6 +91,11 @@ const onDataManagedCandles = async (instance = {}, candles, meta) => {
     await updateState(instance, { lastCandleShort: lastCandle })
   }
 
+  if (outdated) {
```

Comment:
I think if it's outdated, we should not update the candles at all in the state.

Reply:
Let me know If i understand it correctly and the following logic makes sense? Lets say `sendPing` returned `ts` as 10:03 am and we are subscribed to 5m candle, if i get the candle for 10:00 am we reject it since outdated will be true. But when we receive candle for 10:05 am, we update the `ts` and the long and short candle if matched. Because when its 10:10 am, we still get candle values for 10:05 am and because we aren't updating `ts` right now .. we end up with candle of 10:05 am which is the outdated candle since its already 10:10 am.

Reply:
> I think if it's outdated, we should not update the candles at all in the state. 

well we can pre-feed the cache, but should not trade on the data. the more data points we have, the more accurate our algo is

Reply:
using historic values to get the current ema value is fine, just not trading on them

Reply:
https://www.investopedia.com/ask/answers/122314/what-exponential-moving-average-ema-formula-and-how-ema-calculated.asp